torch/_refs/__init__.py

```diff
@@ -5216,7 +5216,10 @@ def linspace(
         return torch.full((0,), 0, dtype=dtype, **factory_kwargs)  # type: ignore[arg-type]
     if steps == 1:
         if isinstance(start, TensorLikeType):
-            return torch.empty((steps,), dtype=dtype, **factory_kwargs).copy_(start)  # type: ignore[arg-type]
+            # we need to convert tensor(0,) to torch.Size([1])
+            if len(start.shape) == 0:
+                return torch.full((steps,), start.item(), dtype=dtype, **factory_kwargs)  # type: ignore[arg-type]
+            return start.clone()
```

Comment:
the original decomp promised to keep the same metadata as `dtype` and `factory_kwargs`. is that preserved here?

Reply:
you probably just want to use `aten.copy(torch.empty(....))` directly to ensure you get the same semantics

(although we should just fix the functionalization-after-decomp story :) )